### PR TITLE
k8s: add namespace api

### PR DESCRIPTION
**Pull request by rithujohn191**
### Description
Add Kubernetes namespace resource API and describeNamespace API call

### Testing Performed
- Unit test
- Local test:
```
curl -X POST   http://localhost:3000/v1/k8s/describeNamespace -H 'Content-Type: application/json'   -d '{ 
"clientset": "kind-clutch-local",
"cluster": "kind-clutch-local",
"name": "envoy-staging"
}'
{"namespace":{"cluster":"kind-clutch-local", "name":"envoy-staging", "annotations":{}, "labels":{}}}
```

### CODE REVIEW

File: backend/service/k8s/namespace.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if len(namespaces.Items) == 1 {
+		return ProtoForNamespace(cs.Cluster(), &namespaces.Items[0]), nil
+	} else if len(namespaces.Items) > 1 {
+		return nil, fmt.Errorf("Located multiple Namespaces")
+	}
+	return nil, fmt.Errorf("Unable to locate namespaces")
```
Comment (danielhochman) [inline backend/service/k8s/namespace.go:31]: use rich errors `status.Error`

File: api/k8s/v1/k8s.proto [hunk 2]
```diff
@@ -852,6 +860,38 @@ message CreateJobResponse {
...
+  option (clutch.api.v1.id).patterns = {
+    type_url : "clutch.k8s.v1.Namespace",
+    pattern : "{cluster}/{name}"
+  };
+
+  string clientset = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string cluster = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  string name = 3 [ (validate.rules).string = {min_bytes : 1} ];
+
+  map<string, string> labels = 4 [ (validate.rules).map.keys.string = {min_bytes : 1} ];
```
Comment (danielhochman) [inline api/k8s/v1/k8s.proto:886]: what does this field do on the request?

**Review by danielhochman — COMMENTED** [review]
looks good after a quick pass, will take one more look to make sure i didn't miss anything after comments are addressed

File: api/k8s/v1/k8s.proto (hunk 2 above)
Comment (rithujohn191) [reply api/k8s/v1/k8s.proto:886]: You can provide namespace specific labels to filter by just like you can for any other resource

Comment (danielhochman) [reply api/k8s/v1/k8s.proto:886]: i figured that was the intent but when i looked it seems like the field is never read or used for that purpose in the implementation.

Comment (rithujohn191) [reply api/k8s/v1/k8s.proto]: Thanks for pointing out that this is a redundant field for a describe call which is trying to get a specific resource and not list based on labels. I'll clean up the other describe calls in a separate PR as all of them have the label field.

**Review by danielhochman — APPROVED** [review]

**Pull request merged**